src/metabase/api/dashboard.clj

```diff
@@ -182,26 +183,32 @@
 (defn- get-dashboard
   "Get Dashboard with ID."
   [id]
-  (-> (t2/select-one :model/Dashboard :id id)
-      api/read-check
-      ;; i'm a bit worried that this is an n+1 situation here. The cards can be batch hydrated i think because they
-      ;; have a hydration key and an id. moderation_reviews currently aren't batch hydrated but i'm worried they
-      ;; cannot be in this situation
-      (t2/hydrate [:dashcards
-                   [:card [:moderation_reviews :moderator_details]]
-                   :series
-                   :dashcard/action
-                   :dashcard/linkcard-info]
-                  :tabs
-                  :collection_authority_level
-                  :can_write
-                  :param_fields
-                  :param_values
-                  [:collection :is_personal])
-      collection.root/hydrate-root-collection
-      api/check-not-archived
-      hide-unreadable-cards
-      add-query-average-durations))
+  (span/with-span!
+    {:name       "get-dashboard"
+     :attributes {:dashboard/id id}}
+    (let [dashboard (-> (t2/select-one :model/Dashboard :id id) api/check-404 api/read-check)
```

Comment:
the check 404 and read-check are duplicative. read-check has a check 404 inside of it.

Reply:
Fixed